### PR TITLE
test: add comprehensive LogoCarousel unit tests

### LINKED ISSUE #1843: Add tests for <LogoCarousel> component
Write unit tests for the `<LogoCarousel>` React component to ensure expected behavior, edge case handling, and rendering logic.

## Essential Test Coverage Checklist

- [ ] **Renders successfully with minimal required props**  
- [ ] **Conditional rendering logic**  
- [ ] **Prop-based behavior** – different props affect output  
- [ ] **Event handling** – simulate user actions and verify callbacks  
- [ ] **State changes / internal logic**  
- [ ] **Default values and fallbacks**  
- [ ] **Text and content rendering**  
- [ ] **Handles edge cases and invalid inputs**  
- [ ] **Accessibility roles and labels**  
- [ ] **DOM structure / classNames / styles**

## Test Reference  
You can refer to the `AutoScrollToTop.test.tsx` file for an example of structure and best practices.  
To explore more examples, see the full component tests folder.

### PR DESCRIPTION
**Pull request by RizWaaN3024**
## Proposed change
Resolves #1843

This PR adds comprehensive unit tests for the LogoCarousel component to improve test coverage and ensure component reliability.

### What was added:
- **33 unit tests** covering all essential test coverage points
- **Basic rendering tests** - verifies component renders with minimal props and empty arrays
- **Conditional rendering logic** - tests image rendering when imageUrl is present/absent
- **Prop-based behavior** - tests how different sponsor props affect output and animation duration
- **Event handling** - tests link clicks, accessibility attributes, and user interactions
- **State changes and internal logic** - tests useEffect behavior and ref usage for DOM manipulation
- **Default values and fallbacks** - tests handling of missing properties and empty image URLs
- **Text and content rendering** - tests all text content, sponsor names, and alt text
- **Edge cases and invalid inputs** - tests null values, long names, special characters, invalid URLs, and large datasets
- **Accessibility roles and labels** - tests proper alt text, link attributes, and semantic structure
- **DOM structure, classNames, and styles** - tests CSS classes, styling, and DOM hierarchy

### Technical details:
- Properly mocks Next.js `Image` and `Link` components for testing
- Handles component's innerHTML duplication behavior for seamless scrolling
- Uses proper TypeScript types imported from `types/home`
- Follows project's testing patterns and conventions
- Clean, readable code with proper formatting

### Test coverage:
- ✅ Renders successfully with minimal required props
- ✅ Conditional rendering logic
- ✅ Prop-based behavior
- ✅ Event handling
- ✅ State changes / internal logic
- ✅ Default values and fallbacks
- ✅ Text and content rendering
- ✅ Handles edge cases and invalid inputs
- ✅ Accessibility roles and labels
- ✅ DOM structure / classNames / styles

## Checklist

- [x] I've read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md).
- [x] I've run `make check-test` locally; all checks and tests passed.

### CODE REVIEW

File: frontend/__tests__/unit/components/LogoCarousel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+
+  describe('State Changes and Internal Logic', () => {
+    it('duplicates innerHTML on mount and when sponsors change', () => {
+      const { rerender } = render(<MovingLogos sponsors={mockSponsors} />)
+
+      const sponsorLinks = screen
+        .getAllByTestId('sponsor-link')
+        .filter(
+          (link) =>
+            link.getAttribute('href')?.includes('sponsor1.com') ||
```
[the remark this answers is not in the document]
Comment (kasya) [reply frontend/__tests__/unit/components/LogoCarousel.test.tsx]: @RizWaaN3024 You need to address these first. Either resolve them or mark as resolved (i'd prefer you actually updating the code)

**Review by kasya — APPROVED** [review]
These looks good! Thank you 👍🏼
